File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/common/tasks/WaitTask.java
package org.firstinspires.ftc.teamcode.common.tasks;

import org.firstinspires.ftc.teamcode.common.BunyipsOpMode;

public class WaitTask extends Task implements TaskImpl {
    public WaitTask(BunyipsOpMode opMode, double time) {
        super(opMode, time);

    }

    void update() {}

    @Override
    public void run() {
        if (isFinished()) {
            return;
        }
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/common/tasks/WaitTask.kt
package org.firstinspires.ftc.teamcode.common.tasks

import org.firstinspires.ftc.teamcode.common.BunyipsOpMode

/**
 * Task to wait for a specific amount of time.
 */
class WaitTask(opMode: BunyipsOpMode, time: Double) : Task(opMode, time), AutoTask {
    override fun run() {
        opMode.idle()
    }
}